src/mergecal/calendar_merger.py

```diff
@@ -31,16 +29,25 @@ def __init__(
         if method:
             self.merged_calendar.add("method", method)
 
-        self.calendars: list[Calendar] = calendars
+        self.calendars: list[Calendar] = []
+
+        for calendar in calendars:
+            self.add_calendar(calendar)
 
     def add_calendar(self, calendar: Calendar) -> None:
         """Add a calendar to be merged."""
-        self.calendars.append(calendar)
+        self.calendars.append(to_standard(calendar, add_timezone_component=True))
 
     def merge(self) -> Calendar:
         """Merge the calendars."""
-        existing_uids = set()
+        existing_uids: set[tuple[Optional[str], int, Optional[str]]] = set()
+        no_uid_events: list[Event] = []
+        tzids: set[str] = set()
         for cal in self.calendars:
+            for timezone in cal.timezones:
+                if timezone.tz_name not in tzids:
+                    self.merged_calendar.add_component(timezone)
+                    tzids.add(timezone.tz_name)
             for component in cal.events:
```

Comment:
I dont know why we/I choose the variable `component`
I think `event` might be more clear

Same how you doing it with timezone components